### ISSUE TITLE
Extract logits

### ISSUE THREAD

**Issue by bmschmidt** [nomic-ai/pyllamacpp#3]
This is the big one for me. In addition to just predicting words at a temperature, to really hack at this thing we want to get to the logit representations out of the cpp memory state into vectors we can manipulate and save in python.

The way I've been thinking about this is something like a `model.braindump("~/output.bin")` method that captures the embedded representations, token by token, of the current conversation. These are things that we could then be sharing and distributing to compare elements of model state, to debug errors, all sorts of crazy fun stuff.
